From 9bfc06d51a06302e2329c8cc7e2f31e30e2f0af3 Mon Sep 17 00:00:00 2001
From: Glenn Washburn <development@efficientek.com>
Date: Fri, 5 Nov 2021 14:12:56 -0500
Subject: [PATCH 069/261] docs: Update development docs to include information
 on running test suite

Add a section with minimal description on setting up and running the test
suite with a link to the INSTALL documentation which is a little more
detailed in terms of package requirements.

Signed-off-by: Glenn Washburn <development@efficientek.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 docs/grub-dev.texi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/docs/grub-dev.texi b/docs/grub-dev.texi
index fb2cc965e..cd59da66c 100644
--- a/docs/grub-dev.texi
+++ b/docs/grub-dev.texi
@@ -77,6 +77,7 @@ This edition documents version @value{VERSION}.
 * Coding style::
 * Finding your way around::
 * Contributing Changes::
+* Setting up and running test suite::
 * Updating External Code::
 * Porting::
 * Error Handling::
@@ -483,6 +484,17 @@ If your intention is to just get started, please do not submit a inclusion
 request. Instead, please subscribe to the mailing list, and communicate first
 (e.g. sending a patch, asking a question, commenting on another message...).
 
+@node Setting up and running test suite
+@chapter Setting up and running test suite
+
+GRUB is basically a tiny operating system with read support for many file
+systems and which has been ported to a variety of architectures. As such, its
+test suite has quite a few dependencies required to fully run the suite.
+These dependencies are currently documented in the
+@uref{https://git.savannah.gnu.org/cgit/grub.git/tree/INSTALL, INSTALL}
+file in the source repository. Once installed, the test suite can be started
+by running the @command{make check} command from the GRUB build directory.
+
 @node Updating External Code
 @chapter Updating external code
 
-- 
2.36.1

